nautobot/core/tests/test_settings_schema.py

```diff
@@ -0,0 +1,297 @@
+        fake = faker.Faker()
+        if value_type == "boolean":
+            return str(fake.boolean())
+        elif value_type == "integer":
+            return str(fake.random_int())
+        return f"FAKE_ENV_{fake.word()}"
+
+    def _parse_environment_variable_for_setting(self, setting_name, setting_schema):
+        """Attempt to parse an environment variable for a setting based on its schema."""
+        setting_type = setting_schema.get("type", "string")
+        env_var = setting_schema["environment_variable"]
+
+        # ALLOWED_HOSTS is a special case (space separated instead of commas)
+        if setting_name == "ALLOWED_HOSTS":
+            return os.environ[env_var].split(" ")
+        elif setting_type == "array":
+            children_type = setting_schema.get("items", {}).get("type", "string")
+            if children_type == "integer":
+                return [int(v) for v in os.environ[env_var].split(",")]
+            else:
+                return os.environ[env_var].split(",")
+        elif setting_type == "boolean":
+            return is_truthy(os.environ[env_var])
+        elif setting_type == "integer":
+            return int(os.environ[env_var])
+        else:
+            return os.environ[env_var]
+
+    def _set_fake_environment_variable(self, setting_name, setting_schema):
+        """Set a fake environment variable for a setting based on its schema."""
+        setting_type = setting_schema.get("type", "string")
+        env_var = setting_schema["environment_variable"]
+
+        # ALLOWED_HOSTS is a special case (space separated instead of commas)
+        if setting_name == "ALLOWED_HOSTS":
+            os.environ[env_var] = " ".join(self._get_fake_env_value("string") for _ in range(3))
+        elif setting_type == "array":
+            children_type = setting_schema.get("items", {}).get("type", "string")
+            os.environ[env_var] = ",".join(self._get_fake_env_value(children_type) for _ in range(3))
+        else:
+            os.environ[env_var] = self._get_fake_env_value(setting_type)
+
+    def test_environment_variable_settings(self):
+        """Check that settings with an environment_variable key are using the environment."""
+        with mock.patch.dict(os.environ, {}):
+            # Set fake environment vars based on defined schema types
+            for name, schema in self.settings_schema["properties"].items():
+                if "environment_variable" in schema:
+                    self._set_fake_environment_variable(name, schema)
+
+            # Force reimport of settings module after mocking out os.environ above
+            sys.modules.pop("nautobot.core.settings", None)
+            import nautobot.core.settings as nautobot_settings
+
+            # Test that settings match the mocked environment
+            for name, schema in self.settings_schema["properties"].items():
+                if name not in nautobot_settings.__dict__.keys():
+                    continue
```

Comment:
What would be the cases here?

Reply:
Django and celery settings that we don't directly namespace in `nautobot.core.settings`

Reply:
I'd think that a lot of those would be caught by the check for an `environment_variable` below, but makes sense. Thanks.

Reply:
It's actually only skipping `CELERY_BROKER_USE_SSL` here

Reply:
An error in settings.yaml actually. We can't support this in an environment variable.

Reply:
Curious why not - let's discuss in standup.

Reply:
Oh, because it's a dict?